lib/battle/CBattleInfoCallback.cpp

```diff
@@ -1199,7 +1186,7 @@ std::pair<const battle::Unit *, BattleHex> CBattleInfoCallback::getNearestStack(
 
 	for(const battle::Unit * st : possible)
 	{
-		for(BattleHex hex : avHexes)
+		for(BattleHex hex : battleGetAvailableHexes(reachability, closest, false))
```

Comment:
Shouldn't old code be a little bit faster since this method is called only once?